tests/model/documentselection.js

```diff
@@ -442,6 +442,50 @@ describe( 'DocumentSelection', () => {
 
 			expect( selection.markers.map( marker => marker.name ) ).to.have.members( [ 'marker' ] );
 		} );
+
+		it( 'should fire change:marker event when selection markers change', () => {
```

Comment:
I don't like that this is a single test. In the code I can see at least 4 or 5 scenarios. `setTo`, `setFocus`, `changed` not being set to `true`, being set to `true` by the first loop and by the second loop. 5 cases.

Reply:
I've split this test into multiple unit tests and added a few more. I hope you will be satisfied ;).